Mathlib/Order/PrimeSeparator.lean

```diff
@@ -0,0 +1,155 @@
+/-
+Copyright (c) 2024 Sam van Gool. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sam van Gool
+-/
+
+import Mathlib.Order.PrimeIdeal
+import Mathlib.Order.Zorn
+
+
+/-!
+# Separating prime filters and ideals
+
+In a bounded distributive lattice, if $F$ is a filter, $I$ is an ideal, and $F$ and $I$ are
+disjoint, then there exists a prime ideal $J$ containing $I$ with $J$ still disjoint from $F$.
+This theorem is a crucial ingredient to [Stone's][Sto1938] duality for bounded distributive
+lattices. The construction of the separator relies on Zorn's lemma.
+
+## Tags
+
+ideal, filter, prime, distributive lattice
+
+## References
+
+* [M. H. Stone, Topological representations of distributive lattices and Brouwerian logics
+(1938)][Sto1938]
+-/
+
+
+universe u
+variable {α : Type*}
+
+open Order Ideal Set
+
+variable [DistribLattice α] [BoundedOrder α]
+
+variable {F : PFilter α} {I : Ideal α}
+
+namespace DistribLattice
+
+
+lemma mem_ideal_sup_principal (a b : α) (J : Ideal α) : b ∈ J ⊔ principal a ↔ ∃ j ∈ J, b ≤ j ⊔ a :=
+  ⟨fun ⟨j, ⟨jJ, _, ha', bja'⟩⟩ => ⟨j, jJ, le_trans bja' (sup_le_sup_left ha' j)⟩,
+    fun ⟨j, hj, hbja⟩ => ⟨j, hj, a, le_refl a, hbja⟩⟩
+
+theorem prime_ideal_of_disjoint_filter_ideal (hFI : Disjoint (F : Set α) (I : Set α)) :
```

Comment:
Since `PFilter` and `Ideal` are extended by changing the preorder relation, is the “dual” result worth being stated?

Reply:
Sure, I added it.

Reply:
I'm not sure exactly what mathematical statement you mean? "If F and I are disjoint there exists a prime filter containing G and disjoint from I"? Yes, mathematically this is trivial to derive from what I proved, but it seems that "prime filter" for a distributive lattice has not been defined yet. (Of course it should! But I'm trying to keep the PR minimal.)

Reply:
So you answered my question mathematically, the rest — if interesting — can be indicated as a TODO in that file, so that if somebody needs it, they will find the remark and maybe find it helpful.